Admin LTE 2.0

Soullivaneuh opened the pull request:
This PR is a work to integrate Admin LTE 2.0 on Sonata as requested in #2753.

This PR is currently on progress and is not stable for the moment.

I open it now to lets you try it and discuss it.

This work is split on 3 PR:
 * This PR for admin-lte 2.0 refactoring
 * sonata-core for some vendors update (https://github.com/sonata-project/SonataCoreBundle/pull/144)
 * sonata-user for the admin login page (https://github.com/sonata-project/SonataUserBundle/pull/527)
 * sonata-doctrine-orm-admin for collections buttons and audit box (https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/431)
 * sandbox for welcome box (https://github.com/sonata-project/sandbox/pull/472)
 * sonata-news for posts and comments boxes (https://github.com/sonata-project/SonataNewsBundle/pull/235)
 * sonata-timeline for timeline box header fix (https://github.com/sonata-project/SonataTimelineBundle/pull/29)
 * ecommerce for orders and customers boxes (https://github.com/sonata-project/ecommerce/pull/326)

Demo: http://sonata.sullivansenechal.com/admin/dashboard

Thanks.

Soullivaneuh commented:
On sandbox, I can see this config for sonata-admin:

```
#    assets:
#        stylesheets:
            # The sandbox includes default pre-optimized version of some css and js
            # So you can either use them, and add your custom assets here
            #   ~ or ~
            # tweak the assetic.yml and the following lines
            - assetic/sonata_admin_css.css
            - assetic/sonata_formatter_css.css
            - assetic/sonata_jqueryui_css.css
            - bundles/sonatademo/css/demo.css
```

Should I take care of that? How to compile this css after upgrade?

rande commented:
@Soullivaneuh Can you adjust the ``.gitignore`` file to ignore non used AdminLTE components ? The idea is to avoid super charge the AdminBundle code. The components can be installed individually with bower inside a projet. The ``.gitignore`` file should also remove all html or documentation file, see the current one for reference: https://github.com/sonata-project/SonataAdminBundle/blob/master/.gitignore

The admin's configuration used package defined in assetic https://github.com/sonata-project/sandbox/blob/2.4-develop/app/config/assetic.yml`` and compile with a makefile task: https://github.com/sonata-project/sandbox/blob/2.4-develop/Makefile#L58-L61

Soullivaneuh commented:
Hmm ok @rande, I updated the vendor just because the old one was already here.

Will work on it, thanks for indications.

Soullivaneuh commented:
I suppose I can delete all bower vendor actually on sonata-admin repo ?

rande commented:
I think so.

rande commented:
Also squash your commits it will be easier to review the PR

Soullivaneuh commented:
Ok will work on it.

Squashed commit is on my plan, it on WIP for now! ;)

Another problem, I'm trying to setup sonata-sandbox with my 3 PR as vcs, but I got this composer issue:

```sh
$ composer up
Loading composer repositories with package information
Updating dependencies (including require-dev)                              
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for sonata-project/admin-bundle dev-admin-lte-2.0 as 2.3.x-dev -> satisfiable by sonata-project/admin-bundle[dev-admin-lte-2.0].
    - Installation request for sonata-project/user-bundle dev-admin-lte-2.0 as 2.2.x-dev -> satisfiable by sonata-project/user-bundle[dev-admin-lte-2.0].
    - sonata-project/admin-bundle 2.3.x-dev requires sonata-project/core-bundle dev-master -> satisfiable by sonata-project/core-bundle[dev-master].
    - sonata-project/user-bundle dev-admin-lte-2.0 requires sonata-project/admin-bundle ~2.3@dev -> satisfiable by sonata-project/admin-bundle[2.3.x-dev, 2.4.x-dev].
    - Can only install one of: sonata-project/core-bundle[dev-master, dev-admin-lte-2.0].
    - Can only install one of: sonata-project/core-bundle[dev-master, dev-admin-lte-2.0].
    - Can only install one of: sonata-project/admin-bundle[2.3.x-dev, dev-admin-lte-2.0].
    - Can only install one of: sonata-project/admin-bundle[2.4.x-dev, dev-admin-lte-2.0].
    - Can only install one of: sonata-project/admin-bundle[dev-admin-lte-2.0, 2.3.x-dev].
    - Can only install one of: sonata-project/admin-bundle[2.4.x-dev, dev-admin-lte-2.0].
    - Installation request for sonata-project/core-bundle dev-admin-lte-2.0 as 2.3.x-dev -> satisfiable by sonata-project/core-bundle[dev-admin-lte-2.0].
```

You can find my composer.json file here: https://github.com/Soullivaneuh/sonata-sandbox/blob/admin-lte-2.0/composer.json

rande commented:
Are you using the 2.4-develop branch ?

Soullivaneuh commented:
Yes, my branch is based on `2.4-develop` as you can see here: https://github.com/Soullivaneuh/sonata-sandbox/network

Soullivaneuh commented:
Not sure, but I think is sonata-user-bundle fault.

As you can see [here](https://github.com/sonata-project/SonataUserBundle/blob/2.2/composer.json#L24), it requires sonata-core `~2.2`.

I will just checkout my user-bundle fork on vendor for the moment.

Yes, it's ugly.

Soullivaneuh commented:
Hmm... now I got this error:

```bash
composer up
Loading composer repositories with package information
Updating dependencies (including require-dev)                              
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for sonata-project/core-bundle dev-admin-lte-2.0 as dev-master -> satisfiable by sonata-project/core-bundle[dev-admin-lte-2.0].
    - sonata-project/ecommerce dev-develop requires sonata-project/core-bundle ~2.2@dev -> satisfiable by sonata-project/core-bundle[2.2.x-dev, 2.3.x-dev].
    - Can only install one of: sonata-project/core-bundle[2.2.x-dev, dev-admin-lte-2.0].
    - Can only install one of: sonata-project/core-bundle[2.3.x-dev, dev-admin-lte-2.0].
    - Can only install one of: sonata-project/core-bundle[dev-admin-lte-2.0, 2.2.x-dev].
    - Can only install one of: sonata-project/core-bundle[2.3.x-dev, dev-admin-lte-2.0].
    - Installation request for sonata-project/ecommerce dev-develop -> satisfiable by sonata-project/ecommerce[dev-develop].
```

I think I will have this problem on many bundle... Is there a way to tell composer to not give a sh*t or dependencies constraints ? :)

Soullivaneuh commented:
@rande : As I can see on you [.gitignore](https://github.com/sonata-project/SonataAdminBundle/blob/master/.gitignore#L12-L69), you ignore all bower installed file except dist file for use it on your project?

Should I keep this work convention or delete and ignore all bower vendor files? I mean ignore `Resources/public/vendor/*`

rande commented:
@Soullivaneuh We need to keep the "dist" vendor files for components used in the AdminBundle. So people without bower can still use it. A

Soullivaneuh commented:
Ok it's done for that.

Soullivaneuh commented:
I think I'm done with Admin LTE 2.0 main integration.

I updated my PR note and added a demo link, can you check this out and give me some review please?

I think we could upgrade the dashboard boxes, would you like me to manage it?

Thanks.

rande commented:
It will be great if you can write a small migration guide to explain the impact or any bc break on the work you have done for the admin bundle. So it will help us to see if we can merge it for this release.

The block need to be migrated too. Can you post a screen shot of the actual state ? Thanks

Soullivaneuh commented:
Here the login page:
![sonata_admin_login](https://cloud.githubusercontent.com/assets/1698357/6463291/474f13ca-c1b1-11e4-96e1-a5ecbef3bd3f.png)

And the dashboard page:
![sonata_admin_dashboard](https://cloud.githubusercontent.com/assets/1698357/6463288/44a54554-c1b1-11e4-8f19-bf44f50a74da.png)

You can see it in action here: http://sonata.sullivansenechal.com/admin/dashboard

For BC break, only one thing: the `sonata_wrapper` block is embedding all dashboard including navbar, according to the [theme's upgrade note](http://almsaeedstudio.com/themes/AdminLTE/documentation/index.html#upgrade).

Where do you want to put this note?

rande commented:
Thanks for the demo, here my feedbacks:

- [ ] the blue notice looks weird and too big ( http://sonata.sullivansenechal.com/admin/sonata/demo/car/list )
- [ ] on the list view the square / list toggle is too close of the right border
- [ ] the page bundle is broken, I think it is only some missing css file but this part is quite critical (http://sonata.sullivansenechal.com/admin/sonata/page/page/8/compose)
- [ ] There is an extra margin on form, see the bad one: http://sonata.sullivansenechal.com/admin/sonata/page/site/2/edit and the good one: http://sonata.sullivansenechal.com/admin/sonata/classification/collection/2/edit
- [ ] [ the list | Add new | Delete ] widget has some issues: buttons's borders are not visible and the text disappear on mouse over, see: http://sonata.sullivansenechal.com/admin/sonata/media/gallery/1/edit?context=default
-[ ] missing background on the history page: http://sonata.sullivansenechal.com/admin/sonata/news/post/2/history

meyerbaptiste commented:
I add this to the list:
- [ ] the ACL editor looks weird: http://sonata.sullivansenechal.com/admin/sonata/news/post/1/acl

Soullivaneuh commented:
@rande, If interested, I recompiled assets for sandbox here: https://github.com/Soullivaneuh/sonata-sandbox/commit/176e434b78126edb7ac68cabea5502eb5a0c8d78

Thanks for your reports, will work on it.

Soullivaneuh commented:
For the weird blue notice [here](http://sonata.sullivansenechal.com/admin/sonata/demo/car/list), it's because of new design definition: http://almsaeedstudio.com/themes/AdminLTE/pages/UI/general.html

What is your proposition? Single text? Another UI element?

Soullivaneuh commented:
Square / list toggle buttons fixed, it was a bad navbar conception which already was on 2.4 branch befor my PR. See https://github.com/sonata-project/SonataAdminBundle/pull/2754/files#diff-01b33b418441737612fb2735e2bc2354R240

For the page compose, I don't have the error if I use compiled asset: http://sonata.sullivansenechal.com/admin/sonata/page/page/8/compose

Maybe a js file of page-bundle not included by default by sonata-admin?

Soullivaneuh commented:
For the extra margin on forms, I think it was already here before my PR.

It's due to a missing `.row` on rendering:

```html
<div class="col-md-12">
    <div class="col-md-6">
```

But, according to the following `SiteAdmin` class. the form group classes are specified by configuration:

```php
protected function configureFormFields(FormMapper $formMapper)
{
    $formMapper
        ->with($this->trans('form_site.label_general'), array('class' => 'col-md-6'))
            // [...]
        ->end()
        ->with($this->trans('form_site.label_seo'), array('class' => 'col-md-6'))
            // [...]
        ->end()
    ;
}
```

So, how should I know when to put a `.row` div?

Thanks.

Soullivaneuh commented:
I managed it with column count.

Please see here: https://github.com/sonata-project/SonataAdminBundle/commit/d64083fed28124b4d191da777fcc0c3d5a9d2d89

Soullivaneuh commented:
Collections action buttons fixed on this PR: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/431

Soullivaneuh commented:
@rande : Is `sonata-ba-view-container` class used for striped table?

If yes, we don't needed anymore as `table-striped` now exists on bootstrap 3.

Can I remove it?

Soullivaneuh commented:
Fixed missing background on show and compare template. Boxes was missing.

Soullivaneuh commented:
@meyerbaptiste : ACL render fixed. Missing boxes too.

I take some time to make some rendering optimization by removing label and specified column sizes.

I think all is done except the weird big blue notice.

Waiting for reviews. :+1:

rande commented:
Hello, here by other feedbacks:

- [ ] the square/list toggle still have the issue in the modal view as it is another template,  http://sonata.sullivansenechal.com/admin/sonata/customer/customer/3/edit and click on list (username field)
- [ ] For the blue notice, I think it is only an issue with the default margin of this box. 
- [ ] it will better to keep ``sonata-ba-view-container`` just to be BC if people is using this class.
- [ ] add a note in the CHANGELOG and UPGRADE-2.4 file about class update:  ``btn-outline`` , ``sonata_wrapper`` and any small change you have done to make the admin/block work with the new theme.

Thanks for this work, the Admin look very good!

Soullivaneuh commented:
> For the blue notice, I think it is only an issue with the default margin of this box.

I don't think so. Callouts have rounded corner on new admin lte theme, so the renderer is weird. Or, we are not talking about the same thing... :)

>  it will better to keep `sonata-ba-view-container` just to be BC if people is using this class.

And why not removing it and notice that on UPGRADE-2.4?

> add a note in the CHANGELOG and UPGRADE-2.4 file about class update: `btn-outline`, `sonata_wrapper` and any small change you have done to make the admin/block work with the new theme.

`btn-outline` removing was done on doctrine-orm-admin, should I put a not here or on sonata-admin?

rande commented:
The blue notice touch the outter box, it will be great to have some space, see screenshot:

![screen shot 2015-03-04 at 13 56 40](https://cloud.githubusercontent.com/assets/14672/6483889/5a7c5b7c-c276-11e4-841a-d4ea58283a75.PNG)

People don't read UPGRADE-2.4 so we need to be BC as much as possible and as the ``SonataAdminBundle`` is the main bundle for admin we need to keep shared information inside this bundle. (Some people read this file ... :)

Soullivaneuh commented:
Not your fault if people don't read the f*cking manual... :p

Soullivaneuh commented:
Not our fault if people don't read the f*cking manual... :p

Soullivaneuh commented:
Not our fault if people doesn't read the f*cking manual... :p

Anymay, so we keep `sonata-ba-view-container`?

rande commented:
@Soullivaneuh yep, keep it

Soullivaneuh commented:
I try to comment out all sonata block on admin config to check out with the new theme but I have the following error for doctrine audit:

```
An exception has been thrown during the rendering of a template ("The block type "sonata.admin_doctrine_orm.block.audit" does not exist") in SonataAdminBundle:Core:dashboard.html.twig at line 109.
```

Any idea?

rande commented:
It is part of the SonataDoctrineOrmBundle, but you need a specific bundle to be enabled

```php
        if (isset($bundles['SimpleThingsEntityAuditBundle'])) {
            $loader->load('audit.xml');

            $container->setParameter('sonata_doctrine_orm_admin.audit.force', $config['audit']['force']);
        }
```

Soullivaneuh commented:
Hmm don't really understand (never used it), where should I put this code?

Another question: Do you have another admin block services not present on demo that I should check?

rande commented:
@Soullivaneuh you need to install and setup ``SimpleThingsEntityAuditBundle`` and the block will be available.

Soullivaneuh commented:
Ok, and for another block service? No easter eggs? :)

rande commented:
@Soullivaneuh nop no easter eggs, or we will fix them ;)

Soullivaneuh commented:
Hmm about the audit block, I setup the following config to test:

```yaml
simple_things_entity_audit:
    audited_entities:
        - FOS\CommentBundle\Entity\Comment
```

And run the following command (not sur if in relation of my config):
```bash
$ ./app/console doctrine:schema:update --dump-sql 
ALTER TABLE order__order CHANGE currency currency CHAR(3) DEFAULT NULL;
ALTER TABLE order__order_audit CHANGE currency currency CHAR(3) DEFAULT NULL;
ALTER TABLE invoice__invoice CHANGE currency currency CHAR(3) NOT NULL;
ALTER TABLE invoice__invoice_audit CHANGE currency currency CHAR(3) DEFAULT NULL;
ALTER TABLE basket__basket CHANGE currency currency CHAR(3) DEFAULT NULL;
$ ./app/console doctrine:schema:update --force
Updating database schema...
Database schema updated successfully! "5" queries were executed
```

But I still have the following error:

```
An exception has been thrown during the rendering of a template ("The block type "sonata.admin_doctrine_orm.block.audit" does not exist") in SonataAdminBundle:Core:dashboard.html.twig at line 109.
```

Soullivaneuh commented:
Ok I missed to enable block on `sonata_block.yml`.

Now I have the following error:

```
An exception has been thrown during the rendering of a template ("An exception occurred while executing 'SELECT e.revtype, body AS body, ancestors AS ancestors, depth AS depth, created_at AS createdAt, state AS state FROM Comment_audit e WHERE e.rev = ?' with params ["1190"]:

SQLSTATE[42S02]: Base table or view not found: 1146 Table 'sonata_sandbox.Comment_audit' doesn't exist") in SonataAdminBundle:Core:dashboard.html.twig at line 109.
```

And if I remove the entity audit from config, this error:

```
An exception has been thrown during the rendering of a template ("Notice: Array to string conversion") in SonataDoctrineORMAdminBundle:Block:block_audit.html.twig at line 39.
```

Any idea?

Soullivaneuh commented:
Find it.

Just changed `{{ changedEntity.id }}` to `{{ changedEntity.id.id }}`, don't know if issue is refered on repo.

Soullivaneuh commented:
Dashboard blocks updated on following PRs:

 * https://github.com/sonata-project/sandbox/pull/472 for welcome box
 * https://github.com/sonata-project/SonataNewsBundle/pull/235 for posts and comments boxes
 * https://github.com/sonata-project/SonataTimelineBundle/pull/29 for timeline box header fix
 * https://github.com/sonata-project/ecommerce/pull/326 for orders and customers boxes
 * https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/431 for audit box

Soullivaneuh commented:
Every think is now fixed and I put an upgrade not.

Please tell me if I miss something.

> The export option overlap: http://sonata.sullivansenechal.com/admin/sonata/classification/tag/list - I think the issue already exist in the current admin, But if you can have a look it will be great ;)

I made a PR about that, but never merged: https://github.com/sonata-project/SonataAdminBundle/pull/2654

The best is to merge this PR first and, if no conflict, merge #2654 just after.

I will manage conflicts if happening.

Thanks.

Soullivaneuh commented:
Conflicts fixed.

rande commented:
@Soullivaneuh there is one issue: http://sonata.sullivansenechal.com/admin/sonata/classification/category/list?hide_context=0&filter%5Bcontext%5D%5Bvalue%5D= the export list still behave badly.

And yes, this look better than the previous version!

adyballa commented:
I tried this Request with our project.
(1) Tried it with SPBowerBundle and ...admin-bundle/bower.json. The v1 still lurks somewhere version makes a conflict. Cant get rid of AdminLTE.js not found
(2) bower install ./vendor/sonata-project/admin-bundle/bower.json does a better job.
But it misses bundles/sonat...admin/css/AdminLTE.css. It does not take the dist-directory.
We have few custom templates.

Soullivaneuh commented:
@adyballa I think bower handle configuration from dist on master branch. This PR is not yet merged so bower get the old assets.

Just use pre-installed assets and it will works fine.

Soullivaneuh commented:
@rande I try your page but I can't see where is the problem. Can I have a screenshot?

Soullivaneuh commented:
I found out an ugly margin-botton on body which let's appear a blank space on bottom.

I removed, was it that?

Soullivaneuh commented:
Oh no ok. Found the bug, dropdown outside the page.

Not sure of how to solve that issue...

Soullivaneuh commented:
We can't change dynamically a dropdown to a "dropup" with twitter bootstrap.

I see 3 solutions:

 1. Use another library/tools for this. Have you some propositions?
 2. Force a bottom-margin on list box to prevent dropdown break. It's very ugly in my opinion.
 3. Deal with it. Because, why not? ;)

What is your thinking @rande?

rande commented:
@Soullivaneuh I merge a few PR from @EmmanuelVella which fix some bootstrap integration with the current theme. However these PR has created some conflicts with your work.

Soullivaneuh commented:
@rande Can you tell me which PR? I will see and fix it.

How about the dropdown issue?

rande commented:
@Soullivaneuh seems to work now ?

rande commented:
The PR from @EmmanuelVella are here: https://github.com/sonata-project/SonataAdminBundle/pulls?q=is%3Apr+is%3Aclosed+author%3AEmmanuelVella

Soullivaneuh commented:
I'm trying to merge without break those another PR, will see it.

rande commented:
@Soullivaneuh just rebase the branch to fix conflict

Soullivaneuh commented:
I got this error:

```
GET http://localhost:9090/bundles/sonatacore/vendor/select2-bootstrap-css/select2-bootstrap.min.css
```

I think the required file (https://github.com/sonata-project/SonataAdminBundle/commit/c365a926f67316ffbe1bdba93681a34e3068fbbf#diff-850942b3ba24ab03a40aaa81b6152852R269) wasn't imported on the PR https://github.com/sonata-project/SonataAdminBundle/pull/2768.

Will do it after fix merge conflicts.

EmmanuelVella commented:
@Soullivaneuh The file was imported in the CoreBundle (see https://github.com/sonata-project/SonataCoreBundle/pull/146/files). Maybe you just need to update it ?

Soullivaneuh commented:
@EmmanuelVella indeed I just saw it. Should be ok now.

Soullivaneuh commented:
I try to make assets for demo after rebased my PR of sonata-admin and sonata-core to be updated to @EmmanuelVella but seems to not takes effect.

Futhermore, I have this graphic issue now, on list footer:

![footer_dec](https://cloud.githubusercontent.com/assets/1698357/6523295/c7ca3248-c3ee-11e4-99a2-9c79911ee9cd.png)

Any idea?

Soullivaneuh commented:
Another error on `/admin/sonata/classification/category/tree?hide_context=0`:

```
Impossible to access an attribute ("context") on a NULL variable ("") in SonataClassificationBundle:CategoryAdmin:list_tab_menu.html.twig at line 4
```

@EmmanuelVella did you modify the `list_tab_menu` template?

EmmanuelVella commented:
After having rebase the AdminBundle and the CoreBundle, the choices should look like this : https://cloud.githubusercontent.com/assets/663607/6517004/292b3648-c397-11e4-9080-1b3a6b50669a.JPG

Did you clear your cache and dump the assets ?

I didn't touch the ``list_tab_menu`` template.

Soullivaneuh commented:
```sh
$ ./app/console cache:clear && make assets
Clearing the cache for the dev environment with debug true
if [ ! -f bin/yuicompressor.jar ]; then curl -L https://github.com/yui/yuicompressor/releases/download/v2.4.8/yuicompressor-2.4.8.jar > bin/yuicompressor.jar; fi;
app/console assets:install --symlink web
Trying to install assets as symbolic links.
Installing assets for Symfony\Bundle\FrameworkBundle into web/bundles/framework
The assets were installed using symbolic links.
Installing assets for Sonata\UserBundle into web/bundles/sonatauser
The assets were installed using symbolic links.
Installing assets for Sonata\PageBundle into web/bundles/sonatapage
The assets were installed using symbolic links.
Installing assets for Sonata\MediaBundle into web/bundles/sonatamedia
The assets were installed using symbolic links.
Installing assets for Ivory\CKEditorBundle into web/bundles/ivoryckeditor
The assets were installed using symbolic links.
Installing assets for Sonata\AdminBundle into web/bundles/sonataadmin
The assets were installed using symbolic links.
Installing assets for Nelmio\ApiDocBundle into web/bundles/nelmioapidoc
The assets were installed using symbolic links.
Installing assets for Sonata\BasketBundle into web/bundles/sonatabasket
The assets were installed using symbolic links.
Installing assets for Sonata\PaymentBundle into web/bundles/sonatapayment
The assets were installed using symbolic links.
Installing assets for Sonata\ProductBundle into web/bundles/sonataproduct
The assets were installed using symbolic links.
Installing assets for Sonata\CommentBundle into web/bundles/sonatacomment
The assets were installed using symbolic links.
Installing assets for Sonata\CoreBundle into web/bundles/sonatacore
The assets were installed using symbolic links.
Installing assets for Sonata\FormatterBundle into web/bundles/sonataformatter
The assets were installed using symbolic links.
Installing assets for Sonata\Bundle\DemoBundle into web/bundles/sonatademo
The assets were installed using symbolic links.
Installing assets for Mopa\Bundle\BootstrapBundle into web/bundles/mopabootstrap
The assets were installed using symbolic links.
app/console assetic:dump
Dumping all dev assets.
Debug mode is off.

11:16:12 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_jqueryui_js.js
11:16:16 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_jqueryui_css.css
11:16:17 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_formatter_js.js
11:16:18 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_formatter_css.css
11:16:19 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_admin_js.js
11:16:24 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_admin_css.css
11:16:32 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_front_js.js
11:16:35 [file+] /home/sullivan/projects/fork/sonata-sandbox/app/../web/assetic/sonata_front_css.css
```

But didn't change anything.

Soullivaneuh commented:
If I don't use compiled assets it works. But still have broken graphic:

![footer_dec_bootstrap](https://cloud.githubusercontent.com/assets/1698357/6523900/7021683a-c3f4-11e4-8aae-f706cf4d8f96.png)

And I don't have any idea for the another error. Any clue @rande?

rande commented:
not really, have you check the css file ordering ?

Soullivaneuh commented:
Generated `head` on my local:

```html
<head>
    <meta charset="UTF-8">
    <meta content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no" name="viewport">

    <link rel="stylesheet" href="/bundles/sonatacore/vendor/bootstrap/dist/css/bootstrap.min.css">
    <link rel="stylesheet" href="/bundles/sonatacore/vendor/components-font-awesome/css/font-awesome.min.css">
    <link rel="stylesheet" href="/bundles/sonatacore/vendor/ionicons/css/ionicons.min.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/vendor/admin-lte/dist/css/AdminLTE.min.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/vendor/admin-lte/dist/css/skins/skin-black.min.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/vendor/iCheck/skins/minimal/minimal.css">
    <link rel="stylesheet" href="/bundles/sonatacore/vendor/eonasdan-bootstrap-datetimepicker/build/css/bootstrap-datetimepicker.min.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/vendor/jqueryui/themes/base/jquery-ui.css">
    <link rel="stylesheet" href="/bundles/sonatacore/vendor/select2/select2.css">
    <link rel="stylesheet" href="/bundles/sonatacore/vendor/select2-bootstrap-css/select2-bootstrap.min.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/vendor/x-editable/dist/bootstrap3-editable/css/bootstrap-editable.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/css/styles.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/css/layout.css">
    <link rel="stylesheet" href="/bundles/sonataadmin/css/tree.css">

    <script>
        window.SONATA_CONFIG = {
            CONFIRM_EXIT: true,
            USE_SELECT2: true,
            USE_ICHECK: true                };
        window.SONATA_TRANSLATIONS = {
            CONFIRM_EXIT:  'You\x20have\x20unsaved\x20changes.'
        };
    </script><style type="text/css"></style>

    <script src="/bundles/sonatacore/vendor/jquery/dist/jquery.min.js"></script>
    <script src="/bundles/sonataadmin/vendor/jquery.scrollTo/jquery.scrollTo.min.js"></script>
    <script src="/bundles/sonatacore/vendor/moment/min/moment.min.js"></script>
    <script src="/bundles/sonatacore/vendor/bootstrap/dist/js/bootstrap.min.js"></script>
    <script src="/bundles/sonatacore/vendor/eonasdan-bootstrap-datetimepicker/build/js/bootstrap-datetimepicker.min.js"></script>
    <script src="/bundles/sonataadmin/vendor/jqueryui/ui/minified/jquery-ui.min.js"></script>
    <script src="/bundles/sonataadmin/vendor/jqueryui/ui/minified/i18n/jquery-ui-i18n.min.js"></script>
    <script src="/bundles/sonataadmin/jquery/jquery.form.js"></script>
    <script src="/bundles/sonataadmin/jquery/jquery.confirmExit.js"></script>
    <script src="/bundles/sonataadmin/vendor/x-editable/dist/bootstrap3-editable/js/bootstrap-editable.min.js"></script>
    <script src="/bundles/sonatacore/vendor/select2/select2.min.js"></script>
    <script src="/bundles/sonataadmin/vendor/admin-lte/dist/js/app.min.js"></script>
    <script src="/bundles/sonataadmin/vendor/iCheck/icheck.min.js"></script>
    <script src="/bundles/sonataadmin/vendor/slimScroll/jquery.slimscroll.min.js"></script>
    <script src="/bundles/sonataadmin/Admin.js"></script>
    <script src="/bundles/sonataadmin/treeview.js"></script>
</head>```

EmmanuelVella commented:
@Soullivaneuh Could you please update your sandbox so I can have a look ?

Soullivaneuh commented:
@EmmanuelVella you can find my sandbox here: https://github.com/Soullivaneuh/sonata-sandbox/tree/admin-lte-2.0

But, you will have to manually clone my forked repositories because it don't works on composer.

Soullivaneuh commented:
I got this graphic issue sometime, but sometime not.

I will try with pull-left method.

Soullivaneuh commented:
Oh, I found why!

```twig
<div class="clearfix">
    <div class="col-md-9">

    </div>
    <div class="pull-right">

    </div>
</div>
```

Ugly... :')

EmmanuelVella commented:
:+1: Good catch !

Soullivaneuh commented:
@EmmanuelVella Wasn't on good view, no ugly col-md tricks here.

But I added a pull-left for the batch actions form, solves issue.

So we still have two bugs:
 
 * /admin/sonata/classification/category/tree?hide_context=0 throws internal error
 * @rande Sorry, but we still have the dropdown outside the window...

Soullivaneuh commented:
Ok category issue is already known: https://github.com/sonata-project/SonataClassificationBundle/issues/86

I found how to fix it, will make a quick PR.

Soullivaneuh commented:
Fixed here: https://github.com/sonata-project/SonataClassificationBundle/pull/92

Can you merge it @rande?

Soullivaneuh commented:
@rande: For croped dropdown, the only solution I found is to add a margin-bottom of 100px on list box if the pager is not present.

See here, without pager: http://sonata.sullivansenechal.com/admin/sonata/classification/category/list?hide_context=0&filter%5Bcontext%5D%5Bvalue%5D=

And here, with a pager: http://sonata.sullivansenechal.com/admin/sonata/customer/customer/list

You can see my fix here: https://github.com/sonata-project/SonataAdminBundle/pull/2754/files#diff-afe5494d385f9cdb1774ed3e28498a19R31

Is that good for you?

rande commented:
@Soullivaneuh Can we use select2 for this widget ? The component solve the issue by detecting itself the correct direction.

Soullivaneuh commented:
Hmm I don't know how to do it but I could try.

But we will lose the button with our icons, not really cool, isn't it?

It's the only graphic bug that was already here. Perhaps could you merge this PR and open another issue about it?

rande commented:
@Soullivaneuh ok, lets add the margin hack for now ;)

Soullivaneuh commented:
It's already added on this PR. ;)

Soullivaneuh commented:
So, is everything OK and ready for merge?

I think so for my part. :+1:

amine26tz commented:
good job @Soullivaneuh ;) can you rebase you PR? you will get some conflict to resolve ..

Soullivaneuh commented:
@amine26tz @rande Thanks. It's done. :+1:

Soullivaneuh commented:
@EmmanuelVella Doh! I found why your select2 bootstrap wasn't shown with compiled assets.

Please see here: https://github.com/Soullivaneuh/sonata-sandbox/commit/b759a165fb032f561d0ab4632e0d41b4e8bf0e43 :)

EmmanuelVella commented:
@Soullivaneuh :+1:

Soullivaneuh commented:
Managing a rebase right now.

Soullivaneuh commented:
Branch rebased, no conflict anymore.

@rande can you, if possible, make this PR as priority? Lot of files is changed and each another merged PR will probably broke mine.

Thanks!

rande commented:
done ;)

Soullivaneuh commented:
Great, thanks! :+1: 

Ping me on issues if you find another missing bugs! ;)

rande commented:
@Soullivaneuh yep, I am reviewing your, @EmmanuelVella and @WouterJ's work ;)

Soullivaneuh commented:
Sonata sandbox is probably not ready for this PR with not compiled assets.

Do you want me to work on it or will you do it manually later?

rande commented:
@Soullivaneuh I will do the work, thanks

rande commented:
@Soullivaneuh I fixed some configuration issues with the sandbox. However, the default theme does not work when it is minified. So I have to use this configuration:

    assets:
        stylesheets:
            - assetic/sonata_admin_css.css
            - assetic/sonata_formatter_css.css
            - assetic/sonata_jqueryui_css.css
            - bundles/sonatademo/css/demo.css
            - bundles/sonataadmin/vendor/admin-lte/dist/css/skins/skin-black.min.css

Which is not that bad as the user can switch theme.

Soullivaneuh commented:
@rande Yes I had to do this too: https://github.com/Soullivaneuh/sonata-sandbox/blob/admin-lte-2.0/app/config/assetic.yml#L116

Btw I also added it on sonata-admin configuration: https://github.com/sonata-project/SonataAdminBundle/blob/master/DependencyInjection/Configuration.php#L263

rande commented:
@EmmanuelVella I think you introduce a regression: with no result the bottom bar (export, batch actions, etc ...) is visible

![screen shot 2015-03-09 at 22 51 25](https://cloud.githubusercontent.com/assets/14672/6565273/29dbfeae-c6af-11e4-9eac-7f710075144d.PNG)

Soullivaneuh commented:
Hmm indeed.

It's probably caused by me, not @EmmanuelVella isn't it?

Why should we see? Just nothing?

rande commented:
@Soullivaneuh just nothing, there is nothing to export when there is not result ;)

Soullivaneuh commented:
@rande I was guessing that but I wanted to be sure. :+1: 

And then, the fix: #2794 ;)

rande commented on Resources/views/CRUD/base_edit_form_macro.html.twig:
```diff
@@ -1,33 +1,53 @@
-                            {% if admin.formfielddescriptions[field_name] is defined %}
-                                {{ form_row(form[field_name])}}
+        {# Manage calculation to determinate if a new row is necessary #}
+        {% set group_class = form_group.class | default('col-md-12') %}
```
@Soullivaneuh we are going to revert this code as it break responsive layout.

Soullivaneuh replied:
@rande how to reproduce this layout break issue? What will be your modifications?

rande replied:
Here the related PR: https://github.com/sonata-project/SonataAdminBundle/pull/2848/files

rande replied:
@Soullivaneuh for instance, in this case https://github.com/sonata-project/SonataNewsBundle/blob/master/Admin/PostAdmin.php#L65 we want the ``Status`` and ``Classification`` on the right side. With your code it is not possible, the last block will be on a new row.

Soullivaneuh replied:
Ok but did you keep in mind the row management? Before my PR we had some margin issue because of non respect of .row/.col bootstrap convention.

Any demo to see the result? Thanks.

rande replied:
@qsomazzi can you reply to @Soullivaneuh ?

qsomazzi replied:
I'll try to explain the issue and add a screenshot : 
In your form, you want:
- 1 big col-md-8 on left
- 3 small col-md-4 on right

With the row system we can only have : 
- 1 row with col-md-8 and 1 col-md-4
- 1 row with 2 col-md-4

It wasn't possible to keep them in the same row

qsomazzi replied:
![capture d ecran 2015-03-25 a 14 13 06](https://cloud.githubusercontent.com/assets/1943676/6825224/387775fa-d2f9-11e4-8948-99cce1d3409b.png)

qsomazzi replied:
@Soullivaneuh here's the demo : http://sonata.qsomazzi.fr/admin

Soullivaneuh replied:
Hmm the margin seems to be good (http://sonata.qsomazzi.fr/admin/app/post/1/edit).

Did you make some modifications since your PR @qsomazzi?

qsomazzi replied:
Yes @Soullivaneuh https://github.com/sonata-project/SonataAdminBundle/pull/2849

Soullivaneuh replied:
Good, seems to be ok for me. :+1:

qsomazzi commented:
![capture d ecran 2015-03-25 a 14 13 06](https://cloud.githubusercontent.com/assets/1943676/6825217/2e5e639e-d2f9-11e4-9e8a-38f8ef1fa691.png)

Soullivaneuh commented:
I agree @qsomazzi but I saw before my PR that sometime we had some `.col` div without a parent `.row`.
This canceled the margin that showed ugly pages and is not bootstrap compliant.

I don't remember where this was produced, that's we I was asking for a demo. ;)

My question is simply did you check if this bug is not back again?

qsomazzi commented:
Indeed, i've not check if this bug and it is back again.
If we add only 1 ``.row`` for all groups, that will do it I think.
I set up a demo and give it soon. @Soullivaneuh

Soullivaneuh commented:
Hmm I forgot to answer on line note... anyway.

I especially make this row calculation because I was thinking that the only way yo solve this problem.

Waiting for the demo, thanks.
